### ISSUE TITLE
Support compiler quering of intel compiler for MPI (mpiicc/mpiicpc)

### ISSUE THREAD

**Issue by hideo67** [microsoft/vscode-cpptools#9410]
### Brief Issue Summary

Hi.

I use an intel compiler "mpiicpc" which is a wrapper around the intel compiler that supplies options for the MPI library.
In vscode I can configure the cmake project so that it successfully builds, but Cmake tools is not getting the full set of include search paths from the compiler, and Intellisense can't find the header files specific to the MPI framework.

I have prepared a minimal source project and console output and logs. (Some of the pathnames have been obfuscated)

The implicit include search paths can be seen by giving "-v" to the compiler, as shown below.  It has a lot of paths that does not show in the debug log.

Does this mean that the specific compiler is not supported?

### Debug Log

```shell
loggingLevel: Debug
loggingLevel has changed to: Debug
cpptools/didChangeCppProperties
$/setTraceNotification
cpptools/fileChanged: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/3.23.1/CompilerIdCXX/CMakeCXXCompilerId.cpp
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeCache.txt
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp
  tag parsing file: /work/00/gt78/t78002/cmake_tools_test/mpi_print_size/build/CMakeFiles/3.23.1/CompilerIdCXX/CMakeCXXCompilerId.cpp
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/Makefile2
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/Makefile.cmake
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/progress.marks
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/cmake.check_cache
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/TargetDirectories.txt
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/3.23.1/CompilerIdCXX/a.out
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/3.23.1/CMakeCXXCompiler.cmake
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeDirectoryInformation.cmake
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/3.23.1/CMakeDetermineCompilerABI_CXX.bin
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/3.23.1/CMakeCXXCompiler.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/3.23.1/CompilerIdCXX/a.out
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeCache.txt
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/CMakeDirectoryInformation.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/Makefile2
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/TargetDirectories.txt
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/DependInfo.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/build.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/cmake_clean.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/compiler_depend.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/compiler_depend.ts
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/depend.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/flags.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/link.txt
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/progress.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmake.check_cache
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/progress.marks
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeLists.txt
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/Makefile
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/cmake_install.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/CMakeCXXCompilerABI.cpp.o
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles/cmTC_64540.dir/CMakeCXXCompilerABI.cpp.o.d
Custom browse configuration received: {
  "browsePath": [
    "/home/my_account/cmake_tools_test/mpi_print_size"
  ],
  "compilerPath": "/opt/some_dir/intel/impi/2019.9.304/intel64/bin/mpiicpc",
  "compilerArgs": [
    "-g"
  ]
}
cpptools/clearCustomConfigurations
cpptools/didChangeCustomBrowseConfiguration
Attempting to get defaults from C++ compiler in "compilerPath" property: '/opt/some_dir/intel/impi/2019.9.304/intel64/bin/mpiicpc'
cpptools/resumeParsing
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/.cmake/api/v1/reply/cache-v2-943a504ce12823e45424.json
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/.cmake/api/v1/reply/index-2022-06-05T13-19-42-0502.json
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeCache.txt
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/3.23.1/CMakeDetermineCompilerABI_CXX.bin
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeDirectoryInformation.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/Makefile.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/Makefile2
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/TargetDirectories.txt
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/cmake.check_cache
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/DependInfo.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/build.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/cmake_clean.cmake
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/compiler_depend.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/compiler_depend.ts
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/depend.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/flags.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/link.txt
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/mpi_print_size.dir/progress.make
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/progress.marks
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/Makefile
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/build/compile_commands.json
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/Makefile
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeFiles
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeCache.txt
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/CMakeLists.txt
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles/CMakeTmp/cmake_install.cmake
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/.cmake/api/v1/reply/cache-v2-c896238dafaad599760c.json
cpptools/fileDeleted: /home/my_account/cmake_tools_test/mpi_print_size/build/.cmake/api/v1/reply/index-2022-06-05T13-19-11-0500.json
cpptools/fileCreated: /home/my_account/cmake_tools_test/mpi_print_size/cpp_debug_log
```

### Additional Information

CMakeLists.txt
```
cmake_minimum_required(VERSION 3.10)
project(mpi_print_size CXX)

add_executable(mpi_print_size mpi_print_size.cpp)
```

source: mpi_print_size.cpp (taken from an public tutorial I wrote)
```
#include <mpi.h>
#include <iostream>
#include <cstdlib>

int main(int argc, char *argv[])
{
    MPI_Init(&argc, &argv);

    int num_procs;
    int my_rank;

    MPI_Comm_size(MPI_COMM_WORLD, &num_procs);
    MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);

    std::cout << "Number of processes : " << num_procs << std::endl;
    std::cout << "My rank number : " << my_rank << std::endl;

    MPI_Finalize();
    return EXIT_SUCCESS;
}
```

output of cmake configuratiion
```
[main] Configuring folder: mpi_print_size 
[driver] Removing /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeCache.txt
[driver] Removing /home/my_account/cmake_tools_test/mpi_print_size/build/CMakeFiles
[proc] Executing command: /some_path/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_CXX_COMPILER:FILEPATH=/opt/some_dir/intel/impi/2019.9.304/intel64/bin/mpiicpc -S/home/my_account/cmake_tools_test/mpi_print_size -B/home/my_account/cmake_tools_test/mpi_print_size/build -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] -- The CXX compiler identification is Intel 19.1.3.20200925
[cmake] -- Detecting CXX compiler ABI info
[cmake] -- Detecting CXX compiler ABI info - done
[cmake] -- Check for working CXX compiler: /opt/some_dir/intel/impi/2019.9.304/intel64/bin/mpiicpc - skipped
[cmake] -- Detecting CXX compile features
[cmake] -- Detecting CXX compile features - done
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: /home/my_account/cmake_tools_test/mpi_print_size/build
```

output of "mpiicpc -v -c ../mpi_print_size.cpp"
```
mpiicpc for the Intel(R) MPI Library 2019 Update 9 for Linux*
Copyright 2003-2020, Intel Corporation.
icpc version 19.1.3.304 (gcc version 4.8.5 compatible)
/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/bin/intel64/mcpcom    --target_efi2 --lang=c++ -_g -mP3OPT_inline_alloca -D__ICC=1910 -D__INTEL_COMPILER=1910 -D__INTEL_COMPILER_UPDATE=3 -D__PTRDIFF_TYPE__=long "-D__SIZE_TYPE__=unsigned long" -D__WCHAR_TYPE__=int "-D__WINT_TYPE__=unsigned int" "-D__INTMAX_TYPE__=long int" "-D__UINTMAX_TYPE__=long unsigned int" -D__LONG_MAX__=9223372036854775807L -D__QMSPP_ -D__OPTIMIZE__ -D__NO_MATH_INLINES -D__NO_STRING_INLINES -D__GNUC_GNU_INLINE__ -D__GNUG__=4 -D__GNUC__=4 -D__GNUC_MINOR__=8 -D__GNUC_PATCHLEVEL__=5 -D__LP64__ -D_LP64 -D_GNU_SOURCE=1 -D__DEPRECATED=1 -D__GXX_WEAK__=1 -D__GXX_ABI_VERSION=1002 "-D__USER_LABEL_PREFIX__= " -D__REGISTER_PREFIX__= -D__INTEL_RTTI__ -D__EXCEPTIONS=1 -D__unix__ -D__unix -D__linux__ -D__linux -D__gnu_linux__ -B -Dunix -Dlinux "-_Asystem(unix)" -D__ELF__ -D__x86_64 -D__x86_64__ -D__amd64 -D__amd64__ "-_Acpu(x86_64)" "-_Amachine(x86_64)" -D__INTEL_COMPILER_BUILD_DATE=20200925 -D__INTEL_OFFLOAD -D__MMX__ -D__SSE__ -D__SSE_MATH__ -D__SSE2__ -D__SSE2_MATH__ -D__pentium4 -D__pentium4__ -D__tune_pentium4__ -_k -_8 -_l --has_new_stdarg_support -_a -_b --gnu_version=40805 -_W5 -p --bool -tused -x --openmp_simd --multibyte_chars -mGLOB_diag_suppress_sys --system_preinclude /usr/include/stdc-predef.h -I/opt/some_dir/intel/impi/2019.9.304/intel64/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/mkl/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/ipp/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/mkl/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/tbb/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/daal/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/stdlib --array_section --simd --simd_func --offload_mode=1 --offload_target_names=mic,MIC --offload_unique_string=icpc0592725397UfrADB --bool -mGLOB_em64t=TRUE -mP1OPT_version=19.1-intel64 -mGLOB_diag_file=mpi_print_size.diag -mGLOB_long_size_64 -mGLOB_routine_pointer_size_64 -mP1OPT_print_version=FALSE -mCG_use_gas_got_workaround=F -mP2OPT_align_option_used=TRUE -mGLOB_gcc_version=485 "-mGLOB_options_string=-I/opt/some_dir/intel/impi/2019.9.304/intel64/include -v -c" -mGLOB_cxx_limited_range=FALSE -mCG_extend_parms=FALSE -mGLOB_compiler_bin_directory=/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/bin/intel64 -mGLOB_as_output_backup_file_name=/tmp/icpcm8mJeIas_.s -mIPOPT_activate -mIPOPT_lite -mGLOB_uarch_tuning=0x0 -mGLOB_product_id_code=0x22006d90 -mCG_bnl_movbe=T -mP3OPT_use_mspp_call_convention -mP2OPT_subs_out_of_bound=FALSE -mP2OPT_disam_type_based_disam=2 -mP2OPT_disam_assume_ansi_c -mP2OPT_checked_disam_ansi_alias=TRUE -mGLOB_ansi_alias -mPGOPTI_value_profile_use=T -mP2OPT_il0_array_sections=TRUE -mGLOB_offload_mode=1 -mP2OPT_offload_unique_var_string=icpc0592725397UfrADB -mP2OPT_hlo_level=2 -mP2OPT_hlo -mP2OPT_hpo_rtt_control=0 -mIPOPT_args_in_regs=0 -mP2OPT_disam_assume_nonstd_intent_in=FALSE -mGLOB_imf_mapping_library=/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/bin/intel64/libiml_attr.so -mPGOPTI_gen_threadsafe_level=0 -mIPOPT_lto_object_enabled -mIPOPT_lto_object_value=1 -mIPOPT_obj_output_file_name=mpi_print_size.o -mIPOPT_whole_archive_fixup_file_name=/tmp/icpcwarchgVg1l7 -mGLOB_linker_version=2.27 -mGLOB_linker=ld -mGLOB_driver_tempfile_name=/tmp/icpctempfilewK1MUO -mP3OPT_asm_target=P3OPT_ASM_TARGET_GAS -mGLOB_async_unwind_tables=TRUE -mGLOB_obj_output_file=mpi_print_size.o -mGLOB_source_dialect=GLOB_SOURCE_DIALECT_C_PLUS_PLUS -mP1OPT_source_file_name=../mpi_print_size.cpp -mP1OPT_full_source_file_name=/home/my_account/cmake_tools_test/mpi_print_size/build/../mpi_print_size.cpp -mGLOB_eh_linux ../mpi_print_size.cpp
#include "..." search starts here:
#include <...> search starts here:
 /opt/some_dir/intel/impi/2019.9.304/intel64/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/mkl/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/ipp/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/mkl/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/tbb/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/daal/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/stdlib
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/compiler/include/intel64
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/compiler/include/icc
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/compiler/include
 /usr/include/c++/4.8.5
 /usr/include/c++/4.8.5/x86_64-redhat-linux
 /usr/include/c++/4.8.5/backward
 /usr/local/include
 /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include
 /usr/include/
 /usr/include
End of search list.

```

**Comment by Colengms** [comment]
> I have pasted the C/C++ output channel on this page in the Debug log section. That debug log is not from CMake Tools. It is in fact the C/C++ channel. The query strings do not show up here.

Hi @hideo67 .  The log you pasted appears to start immediately after enabling debug logging, after an  initial attempt to query the compiler had already been attempted.  Could you provide a debug log from immediately after launching VS Code, until after running `CMake: Configure`?

**Comment by sean-mcmanus** [comment]
Yeah, looks like the compiler isn't return the system include paths when queried. Can you check your compiler documentation or use --help to see if there's a way to get it to return the system includes?

It looks like the newer version of mpiic may be using an LLVM front end compiler instead of the EDG compiler, which may fix the issue: https://www.intel.com/content/www/us/en/develop/documentation/mpi-developer-reference-linux/top/command-reference/compiler-commands/compilation-command-options.html .

**Comment by hideo67** [comment]
Hi @sean-mcmanus , I done some experimenting and research.

The closest thing I could get was the "-v" option. It will show all the system include paths, but also some extra info.

Here is the output for mpiicpc.  Note that the header info went to stderr, while the defines went to stdout.
The C compiler for MPI "mpiicc" and the non-mpi standard c++ compiler "icpc" behaves the same.

```
$ mpiicpc -std=c++14 -m64 -v -E -dM -x c++ /dev/null
mpiicpc for the Intel(R) MPI Library 2019 Update 9 for Linux*
Copyright 2003-2020, Intel Corporation.
icpc version 19.1.3.304 (gcc version 4.8.5 compatibility)
/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/bin/intel64/mcpcom    --target_efi2 --lang=c++ -_g -mP3OPT_inline_alloca -D__ICC=1910 -D__INTEL_COMPILER=1910 -D__INTEL_COMPILER_UPDATE=3 -D__PTRDIFF_TYPE__=long "-D__SIZE_TYPE__=unsigned long" -D__WCHAR_TYPE__=int "-D__WINT_TYPE__=unsigned int" "-D__INTMAX_TYPE__=long int" "-D__UINTMAX_TYPE__=long unsigned int" -D__INTEL_CXX11_MODE__ -D__LONG_MAX__=9223372036854775807L -D__QMSPP_ -D__OPTIMIZE__ -D__NO_MATH_INLINES -D__NO_STRING_INLINES -D__GNUC_GNU_INLINE__ -D__STRICT_ANSI__ -D__GNUG__=4 -D__GNUC__=4 -D__GNUC_MINOR__=8 -D__GNUC_PATCHLEVEL__=5 -D__LP64__ -D_LP64 -D_GNU_SOURCE=1 -D__DEPRECATED=1 -D__GXX_WEAK__=1 -D__GXX_ABI_VERSION=1002 "-D__USER_LABEL_PREFIX__= " -D__REGISTER_PREFIX__= -D__INTEL_RTTI__ -D__EXCEPTIONS=1 -D__unix__ -D__unix -D__linux__ -D__linux -D__gnu_linux__ -B "-_Asystem(unix)" -D__ELF__ -D__x86_64 -D__x86_64__ -D__amd64 -D__amd64__ "-_Acpu(x86_64)" "-_Amachine(x86_64)" -D__INTEL_COMPILER_BUILD_DATE=20200925 -D__INTEL_OFFLOAD -D__MMX__ -D__SSE__ -D__SSE_MATH__ -D__SSE2__ -D__SSE2_MATH__ -D__pentium4 -D__pentium4__ -D__tune_pentium4__ -_k -_8 -_l --has_new_stdarg_support -_a -_b -E --gnu_version=40805 -_W5 -p --bool -tused -x --openmp_simd --c++14 --dM --multibyte_chars -mGLOB_diag_suppress_sys --system_preinclude /usr/include/stdc-predef.h -I/opt/some_dir/intel/impi/2019.9.304/intel64/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/mkl/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/ipp/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/mkl/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/tbb/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/daal/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/include -I/opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/stdlib --array_section --simd --simd_func --offload_mode=1 --offload_target_names=mic,MIC --offload_unique_string=icpc07212842897p17vu --bool -mP1OPT_print_version=FALSE -mGLOB_em64t=TRUE -mP1OPT_version=19.1-intel64 -mGLOB_long_size_64 -mGLOB_routine_pointer_size_64 /dev/null
#include "..." search starts here:
#include <...> search starts here:
 /opt/some_dir/intel/impi/2019.9.304/intel64/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/mkl/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/ipp/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/mkl/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/tbb/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/daal/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/include
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/pstl/stdlib
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/compiler/include/intel64
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/compiler/include/icc
 /opt/some_dir/intel/compilers_and_libraries_2020.4.304/linux/compiler/include
 /usr/include/c++/4.8.5
 /usr/include/c++/4.8.5/x86_64-redhat-linux
 /usr/include/c++/4.8.5/backward
 /usr/local/include
 /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include
 /usr/include/
 /usr/include
End of search list.
#define __SIGNED_CHARS__ 1
#define __PRAGMA_REDEFINE_EXTNAME 1
#define __DATE__ "Jun  9 2022"
#define __TIME__ "17:52:35"
#define __STDC__ 1
#define __cilk 200
#define __STDC_HOSTED__ 1

... snip ...


#define _STDC_PREDEF_H 1
#define __STDC_IEC_559__ 1
#define __STDC_IEC_559_COMPLEX__ 1
#define __STDC_ISO_10646__ 201103L
#define __STDC_NO_THREADS__ 1
```

The "-Wp,-v" option that works for gcc , does not work for intel compilers.

I went through the intel compilers option page [here](https://www.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top/compiler-reference/compiler-options/alphabetical-option-list.html) and done a trivial web search for "intel show system include paths" but nothing useful was found.

There are several versions of the compiler suite installed on the computer center I have access to, but the newest one which is dated "2022.1.2" showed basically the same output. It didn't seem to be based on clang.

**Comment by sean-mcmanus** [comment]
Yeah, thanks for the info -- I believe support for compiler querying with -v instead of "-Wp,-v" when we find `__EDG__ 1` in the defines should be easy to add for our next insiders release.

**Comment by hideo67** [comment]
Thanks for putting this on your list.

**Comment by sean-mcmanus** [comment]
Fixed with 1.11.2: https://github.com/microsoft/vscode-cpptools/releases
